From 74bb9e62492bacda372904d30891eb97685e9b0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 15 Nov 2018 18:47:55 +0100
Subject: [PATCH 07/25] ibusManager: Don't pass undefined callback to ibus

Since commit 551e8278416, we don't always pass a callback parameter.
However passing it on as undefined to ibus doesn't work, as gjs doesn't
accept that as a valid callback value and throw an error. As a result,
we can end up with no layout selected in the keyboard menu and an "empty"
indicator. Fix this by explicitly passing null if no callback has been
provided.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/293
---
 js/misc/ibusManager.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/misc/ibusManager.js b/js/misc/ibusManager.js
index 34f198c35..33ad8777e 100644
--- a/js/misc/ibusManager.js
+++ b/js/misc/ibusManager.js
@@ -210,7 +210,7 @@ var IBusManager = new Lang.Class({
         }
 
         this._ibus.set_global_engine_async(id, this._MAX_INPUT_SOURCE_ACTIVATION_TIME,
-                                           null, callback);
+                                           null, callback || null);
     },
 
     preloadEngines(ids) {
-- 
2.20.0

